### ISSUE TITLE
Clear cached account on Jetpack disconnected

### ISSUE THREAD

**Issue by vbelolapotkov** [Automattic/woocommerce-payments#632]
IMO we should clear cached account if we notice Jetpack connection is broken.

**Steps to reproduce**
1. Connect site with Jetpack.
2. Setup account.
3. Disconnect site from Jetpack.
4. Change site URL and Connect with Jetpack with new URL. Blog id should change at this step.
5. Cached account is still used, and account settings displayed correctly (Active, ready to accept payments, etc). However, all other pages are not functioning as well as transactions.

**Expected behaviour**

The user should be prompted to set up a new WC Pay account after step 4.

**Comment by naman03malhotra** [comment]
@achyuthajoy can you please test this once and see if it is relevant as a porter.

**Comment by naman03malhotra** [comment]
@achyuthajoy gentle reminder regarding this, whenever you get some time.

**Comment by RadoslavGeorgiev** [comment]
Moving back to _Triage in progress_ because it needs to be tested before we can prioritize it.

**Comment by RadoslavGeorgiev** [comment]
Took a look at the issue, and I understand it now. What we need to do is listen for Jetpack hooks (the right one can be explored while working on the issue), and just clear the cached account data, forcing a re-fetch and potentially a prompt to connect a new account.

**Comment by vbelolapotkov** [comment]
@anu-rock routing this one to you since account lifecycle has the most expertise with Jetpack connection.

**Comment by anu-rock** [comment]
@vbelolapotkov I have added it to our backlog, but I am unsure of its priority. Since you reported it, did you notice many merchants being affected by the account cache not being cleared on Jetpack disconnection? For the time being, I am assigning it low priority based on the enhancement tag. Feel free to bump it up if necessary.

**Comment by vbelolapotkov** [comment]
> Since you reported it, did you notice many merchants being affected by the account cache not being cleared on Jetpack disconnection?

No, I didn't. Agree with your assessment.
